### DISCUSSION TITLE
Latest scale-related changes completely broke the look on Windows

### DISCUSSION THREAD

**Discussion by tmssngr** [eclipse-platform/eclipse.platform.swt#1570] [General]
With my SWT jars based on I20240605-1800, the GUI looks
<img width="694" alt="2024-11-04 14_38_56-Preferences" src="https://github.com/user-attachments/assets/872b37d0-9583-4a89-81c5-cf65f1156697">
while with the SWT jars based on I20241028-0700 it looks like this:
<img width="601" alt="2024-11-04 14_41_57-Preferences" src="https://github.com/user-attachments/assets/0da4e2d7-5e24-45bf-88c7-c7634531b515">
Where can I find documentation what changes I need to make to make it look correct again?

**Comment by tmssngr** [comment]
Note, that we used `System.getProperties().putIfAbsent("swt.autoScale", "false")` on Windows to have control over all pixels.

**Reply by HeikoKlare** [reply]
Does it look the same if you start the application with VM argument `-Dswt.autoScale=false`? `swt.autoScale` is not intended to be changed at runtime, as it is used in the static initializer of `DPIUtil`. One thing that might have changed is that `DPIUtil` is initialized earlier now, so that programmatically setting the `swt.autoScale` property comes too late now to have proper effect.

**Comment by Phillipus** [comment]
How does it look without the `autoScale` setting? And can you reproduce with a snippet?

**Reply by tmssngr** [reply]
Without the swt.autoScale set to false, it looks better, but then our HiDPI-images are not loaded. Currently, we are loading them using such code
```java
			image = new Image(display, new ImageDataProvider() {
				private ImageData imageData2x;

				@Override
				public ImageData getImageData(int zoom) {
					if (zoom < 150) {
						return imageData1x;
					}

					if (imageData2x == null) {
						imageData2x = getImageDataNotNull(resourceNameWithoutExtension + "-2.png");
					}

					return imageData2x;
				}
			});
```
Debugging revealed that the parameter `zoom` is always `100`, so my 4k monitor is set to 150%.

**Comment by Phillipus** [comment]
The second screenshot shows labels too close to the check boxes. Can you create a snippet that reproduces that (probably need to specify display zoom has to be 150%)?

**Comment by tmssngr** [comment]
Thanks, I've reported bug #1573.

**Reply by HeikoKlare** [reply]
@tmssngr the bug should have been fixed at latest in today's I-Build: I20241107-0510

Would be great if you found the time to eventually reevalute the reported issues with a current build.

**Comment by tmssngr** [comment]
Thanks for the fix. Yes, it resolves this problem. I will report others, too.
